srcfile.go

```diff
@@ -37,6 +40,19 @@ func (f *SrcFile) Position(offset int) Position {
 	if line >= 0 {
 		lineStart = f.lineOffsets[line]
 	}
+
+	row := line + 2
+	col := offset - lineStart + 1
+
+	if f.sourceMap != nil {
+		if _, _, row, col, ok := f.sourceMap.Source(row, col); ok {
+			return Position{
+				Line: row,
+				Col:  col,
+			}
+		}
+	}
+
 	return Position{
 		Line: line + 2,
 		Col:  offset - lineStart + 1,
```

Comment:
The values are already in row and col variables, no need to calculate them again